### DISCUSSION TITLE
How to  Estimate Stellar Mass in CIAGLE?

### DISCUSSION THREAD

**Discussion by LvZF** [mboquien/cigale#113] [Q&A] [answered]
Hi,
I am wondering how the stellar mass is calculated in CIGALE. I have noticed the code in the 100-146 rows of the pcigale.analysis_modules.pdf_analysis.utils.py file. 

![image](https://github.com/user-attachments/assets/24a127ba-8572-496a-91e2-773d244a0456)

So, now my question is: how to understand the function? Can it be simply understood as F_obs/F_model? Is the _compute_scaling function the F_obs/F_model that has taken into account the observational errors and the input parameter errors?

Thanks！
Best wishes！

**Comment by mboquien** [comment]
Hello. In effect that scaling is computed so that it minimizes the χ², so that is not exactly just an F_obs/F_model. The observational uncertainties are taken into account so that each band is correctly weighted. Note that the scaling is specific to each model, so if the grid has 1 million models, 1 million different scalings are computed.

**Reply by LvZF** [reply]
Thanks!I got it!So, estimating stellar mass based on the scaling, right? I used same SSP and SFH to generate CSP. And the CSP obtained differs from the SED given by CIGALE by 1.08. This might be due to my failure to consider the error.

**Reply by mboquien** [reply, marked as answer]
Yes. Each model has a normalized mass (1 Msun formed [`sfh.integrated`], somewhat less still alive, depending on the exact SFH). That mass is then just scaled by the scaling factor to get the final stellar mass for that specific model.

**Reply by LvZF** [reply]
Ok,thank you very much~
